src/lib/degradeableItems.ts

```diff
@@ -285,22 +325,22 @@ export async function degradeItem({
 		updateBankSetting('degraded_items_cost', itemsDeleted);
 
 		if (hasEquipped) {
-			// If its equipped, unequip and delete it.
+			// If its equipped, give user new item, unequip old item and delete it.
 			const gear = { ...user.gear[degItem.setup].raw() };
+			if (degItem.run !== undefined) {
+				const bankValue = degItem.run(user);
+				await user.addItemsToBank({ items: bankValue, collectionLog: false });
+			}
 			gear[item.equipment!.slot] = null;
 			await user.update({
 				[`gear_${degItem.setup}`]: gear
 			});
-			if (degItem.itemsToRefundOnBreak) {
-				await user.addItemsToBank({ items: degItem.itemsToRefundOnBreak, collectionLog: false });
-			}
 		} else if (hasInBank) {
 			// If its in bank, just remove 1 from bank.
-			let itemsToAdd = undefined;
-			if (degItem.itemsToRefundOnBreak) {
-				itemsToAdd = degItem.itemsToRefundOnBreak;
+			if (degItem.run !== undefined) {
+				const itemsToAdd = degItem.run(user); // Call the function to get the actual Bank value
+				await user.transactItems({ itemsToRemove: new Bank().add(item.id, 1), itemsToAdd });
```

Comment:
This is where the bug I mentioned is; notice it only removes the degraded item if there's an item to refund.

This may not always be the case if something degrades into dust, and should be accounted for, even if not used currently.